### ISSUE TITLE
Confused on how to authenticate WooCommerce REST API

### ISSUE THREAD

**Issue by ozuf** [woothemes/woocommerce#11714]
I have generated the consumer key and consumer secret but I don't know how to aply it to authenticate on Postman. This is the screenshot below:

![screenshot from 2016-08-17 14-28-18](https://cloud.githubusercontent.com/assets/18336192/17737918/230a1580-6487-11e6-8efb-86a6ad8ed3f5.png)

The url looks like this: `https://mystore.com/wp-json/wc/v1/coupons?oauth_consumer_key=ck_XXXXXXXXXXXXXXX&oauth_signature_method=HMAC-SHA1&oauth_timestamp=1471440398&oauth_nonce=Qaq8cK&oauth_version=1.0&oauth_signature=XXXXXXXXXXXXXXXXXXXXXXXXXw=`

But this is returning error 401:

```
{
  "code": "woocommerce_rest_cannot_view",
  "message": "Sorry, you cannot list resources.",
  "data": {
    "status": 401
  }
}
```
I have read the [the documentation](http://woothemes.github.io/woocommerce-rest-api-docs/?php#authentication) but it didn't show any examples.

**Comment by claudiosmweb** [comment]
Better use the `Encode Oauth Signature` option, but this error is not related with the oAuth signature/authenticate...

The problem is that the user used to generate the REST API keys don have permissions to read coupons or your give only `write` permissions for the keys.

I just tested POSTMAN and works fine.

**Comment by ozuf** [comment]
@claudiosmweb  I have changed the permission to read and write but no difference. Please can you help you with the explanation of you are doing your postman request? I know I must be doing something wrong.

Also I discovered that when I pass the **consumer_key** and **consumer_key** into the url like this: 

`https://www.mystoreurl.com/wp-json/wc/v1/products?consumer_key=XXXXXX&consumer_secret=XXXX`

It works. But I don't know if passing it directly like this is secure.

**Comment by claudiosmweb** [comment]
Read the docs, it's oAuth1.0a, have a lot of docs of it and specs.

Here an example working with Postman:

![screenshot from 2016-08-19 19-30-21](https://cloud.githubusercontent.com/assets/1264099/17826056/6ae13b0a-6643-11e6-8968-b4e0dc8f9bab.png)

**Comment by ozuf** [comment]
I have finally gotten the solution to this. In case you are having similar problem, this is how I was able to authenticate postmam:

I used https authentication. In postman, instead of using  oAuth1.0 as the authentication, use Basic authentication and pass consumer key as the username. And the password should be consumer secret.

**Comment by vankampenp** [comment]
I had the same problem, eventhough I was using Basic authentication on HTTPS. As the documentation states, some servers have a problem reading the headers properly. Passing the consumer key and secret on the url solved the problem for me.  As far as I know, this is secure, as on HTTPS this cannot be read.

**Comment by lmatab** [comment]
I got it working using this method http://nobil.cc/blog/oauth-in-wp-rest-api/

**Comment by dstep2007** [comment]
+1 for @ghost's answer.  Basic Auth using the Postman UI worked for me

**Comment by SauravGuha** [comment]
Hi,
Using legacy end point
http://solutionsinc.staging.wpengine.com/wc-api/v2
 am able to get orders..
However when with new api
http://solutionsinc.staging.wpengine.com/wp-json/wc/v1

it is giving the topic mentioned error..

Any help will be greatly appreciated

**Comment by SauravGuha** [comment]
Hi,
It seems I have finally figured out the problem with the code. I have solved the issue but I wanted to let people know about the issue. 

The consumersecret  seems to be having a trailing ampersand (&) symbol when being used to generate the signature. It is very weird, and I found the issue while looking at the WooCommerce Code : 

https://github.com/woocommerce/woocommerce/blob/master/includes/api/class-wc-rest-authentication.php#L364

Woot, finally I have figured out this issue and the problem solved.
